### ISSUE TITLE
empty sections in latex output

### ISSUE THREAD

**Issue by MaPaw** [doxygen/doxygen#6896]
This is for the git version of doxygen.
I use doxygen to document fortran code with lualatex.
The latex output contains many empty (subsub)sections, followed by what should be the heading.  It looks like this:
5.7.1 *
Data Types
whereas it should look like this:
Data Types

The problem can be solved by replacing the following code, lines 550–554 in templates/latex/doxygen.sty:

```
\newcommand{\doxysection}[1]{\@ifstar{\begingroup\sloppy\raggedright\section*{#1}\endgroup}{\begingroup\sloppy\raggedright\section{#1}\endgroup}}
\newcommand{\doxysubsection}[1]{\@ifstar{\begingroup\sloppy\raggedright\subsection*{#1}\endgroup}{\begingroup\sloppy\raggedright\subsection{#1}\endgroup}}
\newcommand{\doxysubsubsection}[1]{\@ifstar{\begingroup\sloppy\raggedright\subsubsection*{#1}\endgroup}{\begingroup\sloppy\raggedright\subsubsection{#1}\endgroup}}
\newcommand{\doxyparagraph}[1]{@ifstar{\begingroup\sloppy\raggedright\paragraph*{#1}\endgroup}{\begingroup\sloppy\raggedright\paragraph{#1}\endgroup}}
\newcommand{\doxysubparagraph}[1]{@ifstar{\begingroup\sloppy\raggedright\subparagraph*{#1}\endgroup}{\begingroup\sloppy\raggedright\subparagraph{#1}\endgroup}}
```

with

```
\newcommand{\doxysection}{\@ifstar{\doxysection@a}{\doxysection@b}}
\newcommand{\doxysection@a}[1]{\begingroup\sloppy\raggedright\section*{#1}\endgroup}
\newcommand{\doxysection@b}[1]{\begingroup\sloppy\raggedright\section{#1}\endgroup}
\newcommand{\doxysubsection}{\@ifstar{\doxysubsection@a}{\doxysubsection@b}}
\newcommand{\doxysubsection@a}[1]{\begingroup\sloppy\raggedright\subsection*{#1}\endgroup}
\newcommand{\doxysubsection@b}[1]{\begingroup\sloppy\raggedright\subsection{#1}\endgroup}
\newcommand{\doxysubsubsection}{\@ifstar{\doxysubsubsection@a}{\doxysubsubsection@b}}
\newcommand{\doxysubsubsection@a}[1]{\begingroup\sloppy\raggedright\subsubsection*{#1}\endgroup}
\newcommand{\doxysubsubsection@b}[1]{\begingroup\sloppy\raggedright\subsubsection{#1}\endgroup}
\newcommand{\doxyparagraph}{\@ifstar{\doxyparagraph@a}{\doxyparagraph@b}}
\newcommand{\doxyparagraph@a}[1]{\begingroup\sloppy\raggedright\paragraph*{#1}\endgroup}
\newcommand{\doxyparagraph@b}[1]{\begingroup\sloppy\raggedright\paragraph{#1}\endgroup}
\newcommand{\doxysubparagraph}{\@ifstar{\doxysubparagraph@a}{\doxysubparagraph@b}}
\newcommand{\doxysubparagraph@a}[1]{\begingroup\sloppy\raggedright\subparagraph*{#1}\endgroup}
\newcommand{\doxysubparagraph@b}[1]{\begingroup\sloppy\raggedright\subparagraph{#1}\endgroup}
```

**Comment by albert-github** [comment]
- how does the problem manifest ?
- does the problem also occur with pdflatex ?

Probably the empty section should not be present, so to get to the source of the problem:
Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem?  Please don't add external links as they might not be persistent.

**Comment by MaPaw** [comment]
I have attached a tar.bz2 file.  When you run doxygen Doxyfile inside doxytest and then make in doxytest/doc/latex, it creates the the refman.pdf file.  If you look at page 9, you see a section 3.1.1 * followed by the line Functions/Subroutines.  Compare with page 9 of the file doxytest/doc/refman_corrected.pdf.

**Comment by albert-github** [comment]
Sorry but I don't see the attachment

**Comment by MaPaw** [comment]
In my sent mailbox it is shown at the end of the text I wrote, but I try again below.

**Comment by albert-github** [comment]
I've seen problems before when attachments are sent through an email. Please try to attach the attachment directly on github.

**Comment by MaPaw** [comment]
It didn't allow me to attach the tar.bz2 file, so attached a tar.gz file:

[doxytest.tar.gz](https://github.com/doxygen/doxygen/files/2992269/doxytest.tar.gz)

**Comment by albert-github** [comment]
I had a look at the attached test case

- in the issue it speaks of a LaTeX problem but in the test case I don't see any latex directory
- when enabling ´GENERATE_LATEX´  and running pdflatex in the latex directory I get a pdf file and I see the indicated `*` when running with pdflatex. When applying the patch (directly to the doxygen.sty in the latex directoy) I get with pdflatex and also with lualatex:
```
Chapter 1.
(./namespaces.tex
! Argument of \doxysection@a has an extra }.
<inserted text>
                \par
l.1 \doxysection
                {Packages}
```
but I don't see the extra `}`

do you have an idea where this extra `}` comes from?

**Comment by MaPaw** [comment]
On 21. Mar 2019, at 16:18, albert-github <notifications@github.com> wrote:
> 
> I had a look at the attached test case
> 
> in the issue it speaks of a LaTeX problem but in the test case I don't see any latex directory
the latex directory is generated by running the command doxygen Doxyfile inside the doxytest/doc directory, so it will be doxytest/doc/latex.  GENERATE_LATEX is enabled in doxytest/doc/Doxyfile.
> when enabling ´GENERATE_LATEX´ and running pdflatex in the latex directory I get a pdf file and I see the indicated * when running with pdflatex. When applying the patch (directly to the doxygen.sty in the latex directoy) I get with pdflatex and also with lualatex:
> Chapter 1.
> (./namespaces.tex
> ! Argument of \doxysection@a has an extra }.
> <inserted text>
>                 \par
> l.1 \doxysection
>                 {Packages}
> but I don't see the extra }
> 
> do you have an idea where this extra } comes from?
> 
no, I don’t get this error message.  When I do
cd doxytest/doc; doxygen Doxyfile; cd latex; make
it compiles fine.
Do you get the same error with the file doxytest/doc/doxygen.sty_corrected?

**Comment by albert-github** [comment]
I cannot find the mentioned doxytest/doc/doxygen.sty_corrected but I think I#ve ben hit by the fact that the  code in the question is not set as code ant thus the ´\´ has been removed.

**Comment by MaPaw** [comment]
On 21. Mar 2019, at 16:51, albert-github <notifications@github.com> wrote:
> 
> I cannot find the mentioned doxytest/doc/doxygen.sty_corrected but I think I#ve ben hit by the fact that the code in the question is not set as code ant thus the ´\´ has been removed.
> 
Sorry if I made some implicit assumptions. When I untar the file doxytest.tar.gz with tar -xf doxytest.tar.gz, I get a directory doxytest with sub-directories doc and src.  And the sub-directory doc contains (among others) the file doxygen.sty_corrected and a latex directory.

**Comment by albert-github** [comment]
I think I have to apologize as I made a complete mess of my test directory (due to the fact that I'm not working on my regular machine). My firsts tests look OK (lualatex and pdflatex), but as soon as I'm back on my regular machine I will do some more tests and provide a proposed pull request (will take a little while).

**Comment by albert-github** [comment]
Based on the provided solution in the issue I did some tests and this looks OK,si I've just pushed a proposed patch, pull request #6902

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
